cf-templates/main.yaml

```diff
@@ -241,6 +248,21 @@ Resources:
         PublicSubnet2: !GetAtt Base.Outputs.PublicSubnet2
         PrivateSubnet1: !GetAtt Base.Outputs.PrivateSubnet1
         PrivateSubnet2: !GetAtt Base.Outputs.PrivateSubnet2
+  
+  OpenSearchIngest:
+    Type: AWS::CloudFormation::Stack
+    Properties:
+      TemplateURL: !Sub https://${S3BucketName}.s3.amazonaws.com/assets/osi.yaml
+      Parameters:
+        AOSDomainEndpoint: !GetAtt OpenSearch.Outputs.AOSDomainEndpoint
+        VPC: !GetAtt Base.Outputs.VPC
+        PrivateSubnet1: !GetAtt Base.Outputs.PrivateSubnet1
+        PrivateSubnet2: !GetAtt Base.Outputs.PrivateSubnet2
+        PrivateSubnet3: !GetAtt Base.Outputs.PrivateSubnet3
+        AOSDomainUserName: !GetAtt OpenSearch.Outputs.AOSDomainUserName
```

Comment:
Are you authenticating using username and password? I recommend using role to follow the best practices.

Reply:
Yes, the username is passed from the template, password is stored in KMS. This seems to be an existing pattern in the workshop. I will add an issue to refactor to use role based method.

Reply:
My mistake! I was thinking that you had implemented the pipeline through basic_auth but in the managed service it is not supported. We will need to keep OpenSearch Dashboards authenticated via basic_auth otherwise we will need to add Cognito or some identity provider.